src/main/resources/workerScripts/fromMultiplePriorityQueues.lua

```diff
@@ -16,7 +18,7 @@ for q in queues.gmatch(queues, NEXT_QUEUE_REGEX) do
     local status, queueType = next(redis.call('TYPE', queueName))
     local payload
     if queueType == 'zset' then
-        local firstMsg = redis.call('ZRANGE', queueName, '0', '0')
+	    local firstMsg = redis.call('ZRANGEBYSCORE', queueName, '-inf', now, 'WITHSCORES', 'LIMIT' , '0' , '1')
```

Comment:
Please looks into ClientImpl.doDelayedEnqueue

It's last parameter **_future_** is used as score:
jedis.**_zadd_**(key, **_future_**, jobJson);

Now lets return to **_fromMultiplePriorityQueues.lua_** script.

Current implementation uses 'ZRANGE', queueName, '0', '0' , returning first element in the sorted set. However , delayed tasked having higher score would be also returned by this command . In order to avoid returning delayed task we would like to limit the score by current time.  That's when we use ZRANGEBYSCORE allowing us to limit the score between min and max. in our case **_min_** is **_-inf_** (unlimited) and **_max_** is **_now_** (allowing us to avoid delayed tasks that are still in the future and should not be visible)

The optional WITHSCORES argument makes the command return both the element and its score, instead of the element alone

The optional LIMIT argument is used to only get a one element  ('0' , '1')


https://redis.io/commands/zadd
https://redis.io/commands/zrange
https://redis.io/commands/zrangebyscore